compose/snippets/src/main/java/com/example/compose/snippets/animations/sharedelement/SharedElementsWithNavigationSnippets.kt

```diff
@@ -193,3 +215,85 @@ data class Snack(
     @DrawableRes val image: Int
 )
 // [END android_compose_shared_element_predictive_back]
+
+
+private sealed class Screen {
+    data object Home : Screen()
+    data class Details(val id: Int) : Screen()
+}
+
+@Preview
+@Composable
+fun CustomPredictiveBackHandle() {
+    // [START android_compose_shared_element_custom_seeking]
+    val seekableTransitionState = remember {
+        SeekableTransitionState<Screen>(Screen.Home)
+    }
+    val transition = rememberTransition(transitionState = seekableTransitionState)
+
+    PredictiveBackHandler(seekableTransitionState.currentState is Screen.Details) { progress ->
+        try {
+            progress.collect { backEvent ->
+                // code for progress
+                seekableTransitionState.seekTo(backEvent.progress, targetState = Screen.Home)
```

Comment:
seekTo may be cancelled _as expected_, if `animateTo` or subsequent `seekTo` gets called before the current `seekTo` finishes. See the mutatorMutex shared between these calls: https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/animation/animation-core/src/commonMain/kotlin/androidx/compose/animation/core/Transition.kt;l=507

You may need a try-catch around this seekTo as well to ignore the cancellation from seekTo.